### PR TITLE
Moved build-all related configuration to CreatorDev/ci

### PR DESCRIPTION
**Pull request by abhijit-mahajani-imgtec**
Added patches and configs required for build-all of openwrt to work.
It has been taken from https://github.com/CreatorKit/build

Add openwrt as submodule for ci release with build-all config
Updated Readme.md with instructions for build-all

make openwrt P=creator-platform-all-cascoda.config V=s

### CODE REVIEW

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "openwrt"]
+	path = openwrt
+	url = https://github.com/Creatordev/openwrt.git
```
Comment (nikhil-zinjurde-imgtec) [inline .gitmodules]: Creatordev => CreatorDev

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	$(MAKE) $(SUBMAKEFLAGS) -C $(DIR__OPENWRT) -j$(J)
+endif
+
+# Copy files to build/output/
+copy_openwrt:
+	mkdir -p $(DIR__CI)/output/openwrt/packages
+	cp -rf $(DIR__OPENWRT)/bin/pistachio/packages/* $(DIR__CI)/output/openwrt/packages/
+	cd $(DIR__CI)/output/openwrt/;tar -cvzf packages.tar.gz packages
+	find $(DIR__OPENWRT)/bin/pistachio/ -maxdepth 1 -type f -exec cp {} $(DIR__CI)/output/openwrt/ \;
+
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: do we need this whole "copy_openwrt" section?  The artifacts (including packages) will be available in openwrt/bin/pistachio and can be uploaded directly from there

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	fi
+
+.PHONY: clean_feeds
+clean_feeds: clean_patches
+	cd $(DIR__OPENWRT); \
+	rm -rf .config feeds.conf tmp/ feeds;
+
+.PHONY: clean_binaries
+clean_binaries:
+	rm -rf $(DIR__CI)/output/
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: if output directory not needed, then this can be replaced with bin/pistachio

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: README.md [hunk 4]
```diff
@@ -1,2 +1,22 @@
-# ci
-CI Repository for generating all packages for openwrt
+##  CreatorDev OpenWrt CI Build system for build-all
+
+Creator-Kit requires multiple repositories for building, which are scattered across two GitHub organizations namely CreatorKit, Creatordev, FlowM2M and Cascoda.
+
+### Steps for triggering build-all release of CreatorDev/Openwrt are as follows :-
+
+Create a directory to clone this CI repository which has openwrt has submodule :-
```
Comment (nikhil-zinjurde-imgtec) [inline README.md]: openwrt _as_ submodule

File: README.md [hunk 5]
```diff
@@ -1,2 +1,22 @@
-# ci
-CI Repository for generating all packages for openwrt
+##  CreatorDev OpenWrt CI Build system for build-all
+
+Creator-Kit requires multiple repositories for building, which are scattered across two GitHub organizations namely CreatorKit, Creatordev, FlowM2M and Cascoda.
```
Comment (nikhil-zinjurde-imgtec) [inline README.md]: This is not required in this repo readme

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
Approved & verified

**Pull request merged**